model/labels/regexp.go

```diff
@@ -872,26 +902,44 @@ func optimizeEqualStringMatchers(input StringMatcher, threshold int) StringMatch
 		numValues++
 		return true
 	}
+	prefixCallback := func(matcher *literalPrefixStringMatcher) bool {
```

Comment:
[nit] I would rename `prefixCallback ` to something like `analysePrefixMatcherCallback`